pkg/controller/managedresource/reconciler.go

```diff
@@ -539,6 +539,22 @@ func computeAllScaledObjectKeys(ctx context.Context, c client.Client) (horizonta
 		}
 	}
 
+	gvkArray := []schema.GroupVersionKind{{"apps", "v1", "Deployment"}, {"apps", "v1", "StatefulSet"}, {"apps", "v1", "DaemonSet"}}
+	for _, gvk := range gvkArray {
+		ul := &unstructured.UnstructuredList{}
+		ul.SetGroupVersionKind(gvk)
+		if err := c.List(ctx, ul); err != nil && !meta.IsNoMatchError(err) {
+			return horizontallyScaledObjects, verticallyScaledObjects, fmt.Errorf("failed to list all Unstructured objects: %w", err)
+		}
+
+		for _, u := range ul.Items {
+			if u.GetAnnotations() != nil && u.GetAnnotations()["gardener-resource-manager.gardener.cloud/preserveResources"] == "true" {
```

Comment:
The annotation should be read from the object [here](https://github.com/gardener/gardener-resource-manager/blob/4d38c292feea2ad3d653f824ea659fb81bded192/pkg/controller/managedresource/merger.go#L39) instead of listing all (potentially unrelated) objects of the same kind.
It should be handled similar to the `preserveResources` parameter, see [here](https://github.com/gardener/gardener-resource-manager/blob/4d38c292feea2ad3d653f824ea659fb81bded192/pkg/controller/managedresource/merger.go#L96).

Reply:
Sure, thank you:)